roombapy/roomba.py

```diff
@@ -464,6 +453,13 @@ def update_state_machine(self, new_state=None):
             self.current_state = ROOMBA_STATES["charge"]
 
         else:
+            if self.cleanMissionStatus_phase not in ROOMBA_STATES:
+                self.log.error(
+                    "Can't find state %s in predefined Roomba states, "
+                    "please create a ticket in "
+                    "https://github.com/pschmitt/roombapy repository",
+                    self.cleanMissionStatus_phase,
+                )
             self.current_state = ROOMBA_STATES[self.cleanMissionStatus_phase]
```

Comment:
We should set current_state to `None` here in case we encounter a yet unknown state.

Reply:
Since the new 'chargingerror' is not handled anywhere before, I guess above statement should still be executed as part of the final else for this value? And to avoid exceptions, maybe set current_state to `None` in the preceding IF and short circuit it after log error message so there is no `KeyError` later on?

Reply:
I hope soon we will refactor whole this method. Currently it looks terrible. So, for now it will be enough to fix the problem.